src/screens/ChatScreen.tsx

```diff
@@ -0,0 +1,305 @@
+// ChatScreen.tsx
+import React, {useState, useEffect, useRef} from 'react'; // Import useRef
+import {
+  Image,
+  KeyboardAvoidingView,
+  Platform,
+  SafeAreaView,
+  ScrollView,
+  StyleSheet,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import {useDispatch, useSelector} from 'react-redux';
+import {
+  widthPercentageToDP as wp,
+  heightPercentageToDP as hp,
+} from 'react-native-responsive-screen';
+import {sendMesageDalleAsync, sendMessageAsync} from '../redux/thunks';
+import {RootState} from '../redux/store';
+
+const ChatScreen: React.FC = () => {
+  const dispatch = useDispatch();
+  const messages = useSelector((state: RootState) => state.chat.messages);
+  const [inputText, setInputText] = useState<string>('');
+  const [isDalleActive, setIsDalleActive] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const scrollViewRef = useRef<ScrollView>(null); // Create a ref for ScrollView
+
+  const handleSendMessage = () => {
+    if (inputText.trim() === '') {
+      return;
+    }
+
+    // Dispatch user message as an object
+    isDalleActive
+      ? dispatch(
+          sendMesageDalleAsync(
+            {role: 'user', content: inputText},
+            setIsLoading,
+          ),
+        )
+      : dispatch(
+          sendMessageAsync({role: 'user', content: inputText}, setIsLoading),
+        );
+    setInputText('');
+  };
+
+  // const renderMessages = () => {
+  //   messages.map((message, index) => {
+  //     if (message.content.startsWith('dalleImage')) {
+  //     }
+  //   });
+  //   return messages.map((message, index) => (
+  //     <View
+  //       key={index}
+  //       style={
+  //         message.role === 'user' ? styles.userMessage : styles.botMessage
+  //       }>
+  //       <Text
+  //         style={[
+  //           styles.messageText,
+  //           {color: message.role === 'user' ? '#fff' : '#004780'},
+  //         ]}>
+  //         {message.content}
+  //       </Text>
+  //     </View>
+  //   ));
+  // };
+
+  useEffect(() => {
+    // Scroll to the bottom when messages change
+    if (scrollViewRef.current) {
+      scrollViewRef.current.scrollToEnd({animated: true});
+    }
+  }, [messages]);
+
+  return (
```

Comment:
You are returning the entire view here, Which is too long code. Can we identify if there is any other approach?